src/sentry/eventstream/kafka/protocol.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+
+import pytz
+import logging
+from datetime import datetime
+
+from sentry.models import Event
+from sentry.utils import json
+
+
+logger = logging.getLogger(__name__)
+
+
+def handle_version_1_message(operation, event_data, task_state):
+    if operation != 'insert':
+        logger.debug('Skipping unsupported operation: %s', operation)
+        return None
```

Comment:
This should maybe explicitly ignore deletes and raise if it's a totally unexpected operation. Also need to verify that delete operations will contain all three items in the function signature

Reply:
Made the task state member optional on delete operations per https://github.com/getsentry/snuba/blob/d0c629c77d77423016d0859f29684c76c4122a37/tests/test_processor.py#L103-L108 — not sure where that is actually published from, though